lib/rpc/SingleJsonRpcProvider.ts

```diff
@@ -313,18 +332,23 @@ export class SingleJsonRpcProvider extends StaticJsonRpcProvider {
       perf.latencyInMs = Date.now() - perf.startTimestampInMs
       this.checkLastCallPerformance(perf)
       this.updateHealthyStatus()
-      if (this.enableDbSync && !this.syncingDb) {
-        this.syncingDb = true
-        // Fire and forget. Won't check the sync result.
-        this.maybeSyncAndUpdateProviderState()
+      if (this.enableDbSync) {
+        if (
+          (!this.syncingDb && this.hasEnoughWaitSinceLastDbSync(1000 * this.config.DB_SYNC_INTERVAL_IN_S)) ||
```

Comment:
do we really still need `syncingDb`? Second boolean condition evaluates to true just purely based on the time lag.

Reply:
You are right. I think I'm being too protective here at the cost of DB access flooding when creating a new lambda (which isn't good). I removed this logic.